pkgs/top-level/all-packages.nix

```diff
@@ -25604,7 +25604,9 @@ with pkgs;
 
   octomap = callPackage ../development/libraries/octomap { };
 
-  odin = callPackage ../development/compilers/odin { };
+  odin = callPackage ../development/compilers/odin {
+    MacOSX-SDK = if darwin.apple_sdk ? sdk then darwin.apple_sdk.sdk else darwin.apple_sdk.MacOSX-SDK;
```

Comment:
This needs an explanation.

Reply:
In my testing it seemed that Odin requires the `sysroot` flag pointing at the apple sdk to compile correctly on macos. Currently there are two `apple_sdk` definitions in nixpkgs:

-   https://github.com/NixOS/nixpkgs/blob/3fc89747ed3abb658cef013c3345b22f4ed99b2e/pkgs/os-specific/darwin/apple-sdk/default.nix#L5
    here it is called `sdk`
- https://github.com/NixOS/nixpkgs/blob/3fc89747ed3abb658cef013c3345b22f4ed99b2e/pkgs/os-specific/darwin/apple-sdk-11.0/default.nix#L23
    here it is called `MacOSX-SDK`

hence why I needed to check if `sdk` exists, otherwise use `MacOSX-SDK`

Reply:
You are sure we don't want to use ``darwin.apple_sdk_11_0.callPackage`` here?

Reply:
Honestly, I have no idea how to make packages compile on Darwin. It also seems that odin wants to link to the System framework now, which is near impossible on nix it seems: https://discourse.nixos.org/t/ld-framework-not-found-system/15096

I'm tempted to make this update linux-only and let Darwin support be added by someone else. I don't really have a MacBook to test this myself anyway.

Reply:
Managed to get a macos virtual machine running -- will try `darwin.apple_sdk_11_0.callPackage` now